enterprise/api_client/discovery.py

```diff
@@ -72,6 +79,36 @@ def __init__(self, user):
         self.user = user
         self.client = course_discovery_api_client(user)
 
+    def get_search_results(self, querystring=None):
+        """
+        Return a search result from all data.
+
+        Returns:
+            dict: API response with search results
+
+        """
+        return self._load_data(
+            self.SEARCH_ALL_ENDPOINT,
+            default=[],
+            querystring=querystring,
+        )
+
+    def get_paginated_search_results(self, querystring=None):
+        """
+        Return paginated search result from all data.
+
+        Returns:
+            dict: API response with search results, as well as links to next and previous pages.
+
+        """
+        return self._load_data(
+            self.SEARCH_ALL_ENDPOINT,
+            default=[],
+            querystring=querystring,
+            traverse_pagination=False,
```

Comment:
@itsjeyd Looks like my original idea was right:

`traverse_pagination (bool): Whether to traverse pagination or return paginated response..`

That's from the edx-platform's `openedx/core/lib/edx_api_utils.py.get_edx_api_data` function's docstring.